packages/subcommands/src/register.ts

```diff
@@ -0,0 +1,52 @@
+import { Plugin, postInitialization, postLogin, SapphireClient } from '@sapphire/framework';
+import { SubCommandsPluginOptions, SubCommandsHandler, SubCommandOptionsResolvable } from './lib/utils/SubCommandsHandler';
+import { join } from 'path';
+import type { ClientOptions } from 'discord.js';
+
+export class SubCommands extends Plugin {
+	public static [postInitialization](this: SapphireClient, options: ClientOptions): void {
+		this.subCommandsHandler = new SubCommandsHandler(options.subCommands);
+		this.events.registerPath(join(__dirname, 'events'));
+	}
+
+	// eslint-disable-next-line @typescript-eslint/no-unused-vars
+	public static [postLogin](this: SapphireClient, _options: ClientOptions): void {
+		this.subCommandsHandler.inspectSubCommandsTree(this.commands);
+	}
+}
+
+declare module '@sapphire/framework' {
+	export interface CommandOptions {
+		subCommands?: SubCommandOptionsResolvable[] | SubCommandOptionsResolvable;
+	}
+
+	export interface Command {
+		subCommands: SubCommandOptionsResolvable[];
+		isSubcommand: boolean;
+	}
+
+	export interface CommandContext {
+		parentName?: string;
+	}
+
+	export interface SapphireClientOptions {
+		subCommands?: SubCommandsPluginOptions;
+	}
+
+	export interface SapphireClient {
+		subCommandsHandler: SubCommandsHandler;
+	}
+}
+
+declare module 'discord.js' {
```

Comment:
Since this augments discord.js, the file should be renamed to `register-discord.js`, or w/e it is with other plugins

Reply:
Ps. might wanna make this a trend. Cba to have things like "register-discord-js.js" and "register-discord.js" and all kinds of possibilities. Just to make sure all plugins are used and behave the same